### ISSUE TITLE
Add 'fetching index' message to reporter

### ISSUE THREAD

**Issue by foresterre** [foresterre/cargo-msrv#120]

**Comment by foresterre** [comment]
When we run `cargo msrv --min 2018 --bisect --output-format json` for this repo, we currently see:

```
{"reason":"mode","mode":"determine-msrv","toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
{"reason":"checking","version":"1.43.1","step":0,"total":25,"toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
{"reason":"installing","version":"1.43.1","step":0,"total":25,"toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
{"reason":"checking","version":"1.43.1","step":0,"total":25,"toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
{"reason":"check-complete","version":"1.43.1","step":0,"total_steps":25,"success":false,"toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
{"reason":"checking","version":"1.50.0","step":1,"total":25,"toolchain":"x86_64-pc-windows-msvc","check_cmd":"cargo check --all"}
... (more messages)
```

When using the default 'human' output format, we similarly start with a message about the mode.
However, before this happens, we fetch an index from a specific source (from #114, the following will be possible: `--release-source rust-dist` or `--release-source rust-changelog` (default)).

We should report the "fetching of the index" to the user, as it may take up some noticeable portion of time, especially with a slower Internet connection.

**Comment by pjsier** [comment]
I'm interested in working on this! Should this call `eprintln!` like some other messages or be implemented in the output formats?

**Comment by foresterre** [comment]
> I'm interested in working on this! Should this call `eprintln!` like some other messages or be implemented in the output formats?

This would be implemented using the output formats (the reporter module).

**Comment by foresterre** [comment]
> 
> 
> Great! From taking an initial look, it seems like the output formats are initialized while running the commands after the index has been fetched:
> 
> https://github.com/foresterre/cargo-msrv/blob/d1fac5955af9eb696f0e09d9863abaee75ca52c5/src/lib.rs#L73-L92
> 
> Would it make sense to create a separate output formatter with a single step at the beginning of the `run_app` method? Or is there a better place for this to plug in?

The reporter should be initialized further at the start of the program. I've implemented this in #126. With #126 it should be possible to use the reporter prior and after the index processing.
